<launch>


    <!-- dataset name -->
    <!-- euroc_V1_01_easy, tum_corridor1_512_16_okvis, udel_gore, udel_arl -->
    <arg name="dataset" default="udel_arl" />
    <arg name="seed"    default="1" />

    <!-- saving to file -->
    <arg name="save_to_file"        default="false" />
    <arg name="stats_path_states"   default="/tmp/vicon2gt_state.csv" />
    <arg name="stats_path_gt"       default="/tmp/vicon2gt_state_gt.csv" />
    <arg name="stats_path_info"     default="/tmp/vicon2gt_info.txt" />

    <!-- vicon noise values -->
    <arg name="vicon_sigmas" default="[0.005,0.005,0.005,0.02,0.02,0.02]" />

    <!-- set the total number of OpenMP threads -->
    <env name="OMP_NUM_THREADS" value="4" />

    <!-- MAIN NODE -->
    <node name="run_simulation" pkg="vicon2gt" type="run_simulation" output="screen" clear_params="true" required="true">

        <!-- simulation -->
        <param name="sim_seed"          type="int"    value="$(arg seed)" />
        <param name="sim_traj_path"     type="string" value="$(find vicon2gt)/data/$(arg dataset).txt" />
        <param name="sim_freq_imu"      type="double" value="200" />
        <param name="sim_freq_cam"      type="double" value="20" />
        <param name="sim_freq_vicon"    type="double" value="100" />
        <!-- how many hz can vicon freq vary-->>

        <!-- save information -->
        <param name="state_freq"           type="int"    value="20" />
        <param name="save_to_file"         type="bool"   value="$(arg save_to_file)" />
        <param name="stats_path_states"    type="string" value="$(arg stats_path_states)" />
        <param name="stats_path_states_gt" type="string" value="$(arg stats_path_gt)" />
        <param name="stats_path_info"      type="string" value="$(arg stats_path_info)" />

        <!-- world parameters -->
        <rosparam param="R_BtoI">[1, 0, 0, 0, 1, 0, 0, 0, 1]</rosparam>
        <rosparam param="p_BinI">[0, 0, 0]</rosparam>
        <rosparam param="R_GtoV">[1, 0, 0, 0, 1, 0, 0, 0, 1]</rosparam>
        <param name="gravity_magnitude"          type="double" value="9.81" />
        <param name="toff_imu_to_vicon"          type="double" value="0.0" />
        <param name="num_loop_relin"             type="int"    value="0" />
        <param name="estimate_toff_vicon_to_imu" type="bool"   value="false" />
        <param name="estimate_ori_vicon_to_imu"  type="bool"   value="true" />
        <param name="estimate_pos_vicon_to_imu"  type="bool"   value="true" />

        <!-- vicon sigmas, only used if we don't get odometry -->
        <!-- sigmas: (rx,ry,rz,px,py,pz) -->
        <rosparam param="vicon_sigmas" subst_value="true">$(arg vicon_sigmas)</rosparam>
        <param name="freq_pub_raw_vicon" type="double" value="10.0" />

        <!-- vi-sensor -->
        <param name="gyroscope_noise_density"      type="double"   value="1.6968e-04" />
        <param name="gyroscope_random_walk"        type="double"   value="1.9393e-05" />
        <param name="accelerometer_noise_density"  type="double"   value="2.0000e-3" />
        <param name="accelerometer_random_walk"    type="double"   value="3.0000e-3" />


    </node>


</launch>